test/Kestrel.Transport.BindTests/HandleInheritanceTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class HandleInheritanceTest : TestApplicationErrorLoggerLoggedTest
+    {
+        [Fact]
+        public async Task SpawnChildProcess_DoesNotInheritListenHandle()
+        {
+            var hostBuilder = TransportSelector.GetWebHostBuilder()
+                .UseKestrel()
+                .ConfigureServices(AddTestLogging)
+                .UseUrls("http://127.0.0.1:0")
+                .Configure(app =>
+                {
+                    app.Run(context =>
+                    {
+                        return context.Response.WriteAsync("Hello World");
+                    });
+                });
+
+            using (var host = hostBuilder.Build())
+            {
+                await host.StartAsync();
+
+                var processInfo = new ProcessStartInfo
+                {
+                    FileName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "cmd.exe" : "sh",
+                    CreateNoWindow = true,
+                };
+                using (var process = Process.Start(processInfo))
+                {
+                    var port = host.GetPort();
+                    await host.StopAsync();
+
+                    using (var client = new TcpClient())
+                    {
+                        await Assert.ThrowsAnyAsync<SocketException>(() => client.ConnectAsync("127.0.0.1", port));
+                    }
+
+                    process.Kill();
```

Comment:
```
08:42:31   Error Message:
08:42:31    System.ComponentModel.Win32Exception : No such process
08:42:31   Stack Trace:
08:42:31      at System.Diagnostics.Process.Kill()
08:42:31      at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.HandleInheritanceTest.SpawnChildProcess_DoesNotInheritListenHandle() in /_/test/Kestrel.Transport.BindTests/HandleInheritanceTest.cs:line 53
08:42:31   --- End of stack trace from previous location where exception was thrown ---
08:42:31   Standard Output Messages:
08:42:31    | [0.010s] TestLifetime Information: Starting test SpawnChildProcess_DoesNotInheritListenHandle at 2018-09-20T15:42:30
08:42:31    | [0.075s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
08:42:31    | [0.173s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
08:42:31    | [0.173s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Libuv.BindTests
08:42:31    | [0.184s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
08:42:31    | [0.238s] TestLifetime Information: Finished test SpawnChildProcess_DoesNotInheritListenHandle in 0.2340523s
```

We need something like this:
https://github.com/aspnet/DotNetTools/blob/60041b5b2bcb84a9a6b9eb460f6e395a270934c8/test/dotnet-watch.FunctionalTests/AwaitableProcess.cs#L146-L157

Reply:
Or maybe this would work? https://github.com/aspnet/Common/blob/e0f40110ff5e2101231ba54d48961bbd60a07727/shared/Microsoft.Extensions.Process.Sources/ProcessHelper.cs

Reply:
On my local run, I see a different error:
```
Failed   SpawnChildProcess_DoesNotInheritListenHandle
Error Message:
 Assert.Throws() Failure
Expected: typeof(System.Net.Sockets.SocketException)
Actual:   (No exception was thrown)
Stack Trace:
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.HandleInheritanceTest.SpawnChildProcess_DoesNotInheritListenHandle() in /Users/jtluo/Documents/workspace/juntaoluo/KestrelHttpServer/test/Kestrel.Transport.BindTests/HandleInheritanceTest.cs:line 50
--- End of stack trace from previous location where exception was thrown ---
Standard Output Messages:
 | [0.016s] TestLifetime Information: Starting test SpawnChildProcess_DoesNotInheritListenHandle at 2018-09-21T03:01:25
 | [0.085s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
 | [0.168s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
 | [0.168s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Sockets.BindTests
 | [0.191s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
 | [0.241s] TestLifetime Information: Finished test SpawnChildProcess_DoesNotInheritListenHandle in 0.2341235s
```
I should take a closer look at what's actually happening in this test.